app/src/main/java/org/fossasia/openevent/app/event/detail/EventDetailsActivity.java

```diff
@@ -50,20 +54,24 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_event_details);
 
-        ButterKnife.bind(this);
+        Event initialEvent = getIntent().getParcelableExtra(EVENT_KEY);
+
+        if(initialEvent == null) {
```

Comment:
what will be the case for this to happen? and `finish` is not good option. It has `event_id`, hence can fetch event from `eventsDataRepository` to show required info.